### PR TITLE
Bump go-git to latest

### PR DESCRIPTION
**Pull request by manno**
Refers to https://github.com/rancher/fleet/issues/2580

### CODE REVIEW

**Review by weyfonk — DISMISSED** [review]

File: go.mod [hunk 1]
```diff
@@ -17,8 +17,8 @@ require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/docker/docker v26.1.4+incompatible
 	github.com/evanphx/json-patch v5.9.0+incompatible
-	github.com/go-git/go-billy/v5 v5.5.0
-	github.com/go-git/go-git/v5 v5.12.0
+	github.com/go-git/go-billy/v5 v5.5.1-0.20240701174852-839dcd3ad096
+	github.com/go-git/go-git/v5 v5.12.1-0.20240629213349-b00c68ab7140
```
Comment (weyfonk) [inline go.mod:21]: Where does this tag live? :eyes:

Comment (manno) [reply go.mod:21]: > vX.Y.(Z+1)-0.yyyymmddhhmmss-abcdefabcdef is used when the base version is a release version like vX.Y.Z. For example, if the base version is v1.2.3, a pseudo-version might be v1.2.4-0.20191109021931-daa7c04131f5.

https://go.dev/ref/mod#pseudo-versions

**Review by weyfonk — DISMISSED** [review]

File: .github/workflows/e2e-ci.yml [hunk 2]
```diff
@@ -103,13 +130,6 @@ jobs:
...
           export CI_OCI_CERTS_DIR="$(git rev-parse --show-toplevel)/FleetCI-RootCA"
 
-          # 1. Run test cases not needing infra
-          ginkgo --github-output --label-filter='!infra-setup' e2e/single-cluster e2e/keep-resources e2e/drift
-
-          # 2. Run tests for metrics
-          ginkgo --github-output e2e/metrics
-          SHARD=shard1 ginkgo --github-output e2e/metrics
-
           # 3. Run tests requiring only the git server
```
Comment (weyfonk) [inline .github/workflows/e2e-ci.yml:133]: On comments 3. to 6.: we should either reset the numbering to start with 1 (or 0 :upside_down_face: ) or remove that numbering altogether. I think the comments still help readability here.
```suggestion
          # Run tests requiring only the git server
```

**Review by weyfonk — APPROVED** [review]
This approach looks good to me, although changes to our end-to-end test workflow _could_ be part of a separate PR.

**Comment by manno** [comment]
> This approach looks good to me, although changes to our end-to-end test workflow _could_ be part of a separate PR.

Sure, I can force merge this PR without the CI changes. I retried it, I think 6 times, yesterday.

File: .github/workflows/e2e-ci.yml [hunk 3]
```diff
@@ -81,17 +86,39 @@ jobs:
...
+        run: |
+          # 1. Run test cases not needing infra
+          ginkgo --github-output --label-filter='!infra-setup && !sharding' e2e/single-cluster e2e/keep-resources e2e/drift
+      -
+        name: E2E Sharding/Metrics Tests
+        if: ${{ matrix.test_type.name == 'sharding' }}
+        env:
+          FLEET_E2E_NS: fleet-local
+        run: |
+          ginkgo --github-output --label-filter='!infra-setup && sharding' e2e/single-cluster
```
[the remark this answers is not in the document]
Comment (manno) [reply .github/workflows/e2e-ci.yml:107]: If that's the case 👍

**Review by weyfonk — APPROVED** [review]

**Pull request merged**